src/lib/Siteaccess/AbstractSiteaccessPreviewVoter.php

```diff
@@ -43,7 +46,15 @@ public function vote(SiteaccessPreviewVoterContext $context): bool
             null,
             $siteaccess
         );
-        if (!in_array($languageCode, $siteaccessLanguages)) {
+        $siteaccessRepository = $this->configResolver->getParameter(
+            'repository',
+            null,
+            $siteaccess
+        );
+        $siteaccessRepository = $siteaccessRepository ?: $this->repositoryConfigurationProvider->pullDefaultRepository();
+        $currentRepository = $this->repositoryConfigurationProvider->getRepositoryConfig()['alias'];
```

Comment:
Could you please move it to private method with proper name in order to improve readability?